src/Template/Bake/config/diff.ctp

```diff
@@ -0,0 +1,492 @@
+                ['unique' => true]
+                <%- endif; %>
+            )
+                        <%- endif;
+                    endforeach;
+                endif;
+
+                foreach($this->Migration->indexes($table) as $index):
+                    sort($foreignKeys);
+                    $indexColumns = $index['columns'];
+                    sort($indexColumns);
+                    if (!in_array($indexColumns, $foreignKeys)):
+                    %>
+            ->addIndex(
+                [<% echo $this->Migration->stringifyList($index['columns'], ['indent' => 5]); %>]<% echo ($index['type'] === 'unique') ? ',' : ''; %>
+
+                <%- if ($index['type'] === 'unique'): %>
+                    ['unique' => true]
+                <%- endif; %>
+            )
+            <%- endif;
+                endforeach; %>
+            ->create();
+            <%- endforeach; %>
+        <%- foreach ($constraints as $table => $tableConstraints):
+            foreach ($tableConstraints as $constraint):
+                $constraintColumns = $constraint['columns'];
+                sort($constraintColumns);
+                if ($constraint['type'] !== 'unique'):
+                    $columnsList = '\'' . $constraint['columns'][0] . '\'';
+                    if (count($constraint['columns']) > 1):
+                        $columnsList = '[' . $this->Migration->stringifyList($constraint['columns'], ['indent' => 5]) . ']';
+                    endif;
+                    $dropForeignKeys[$table][] = $columnsList;
+
+                    if (is_array($constraint['references'][1])):
+                        $columnsReference = '[' . $this->Migration->stringifyList($constraint['references'][1], ['indent' => 5]) . ']';
+                    else:
+                        $columnsReference = '\'' . $constraint['references'][1] . '\'';
+                    endif;
+                    $statement = $this->Migration->tableStatement($table, true);
+                    if (!empty($statement)): %>
+
+        <%= $statement %>
+                    <%- endif; %>
+            ->addForeignKey(
+                <%= $columnsList %>,
+                '<%= $constraint['references'][0] %>',
+                <%= $columnsReference %>,
+                [
+                    'update' => '<%= $constraint['update'] %>',
+                    'delete' => '<%= $constraint['delete'] %>'
+                ]
+            )
+                <%- endif; %>
+            <%- endforeach; %>
+            <%- if (isset($this->Migration->tableStatements[$table])): %>
+            ->update();
+            <%- endif; %>
+        <%- endforeach; %>
+        <%- endif; %>
+        <%- foreach ($data as $tableName => $tableDiff): %>
+            <%- if (!empty($tableDiff['columns']['add'])): %>
+
+        $this->table('<%= $tableName %>')
+            <%- foreach ($tableDiff['columns']['add'] as $columnName => $columnAttributes):
```

Comment:
The few lines above are the FK creations that follows the full table creation.
These part of the template is the creation of columns on tables that already exist.